### PR TITLE
Persistence Component: Improve UDF-based Digest Generation

### PR DESCRIPTION
**Pull request by kumuwu**
#### What type of PR is this?

- Improvement

#### What does this PR do / why is it needed ?

Improve UDF-based Digest Generation. 
1. Expected UDF signature changes from 
`String digestUdf(String[] columnNames, String[] columnValues)` to `String digestUdf(String[] columnNamesAndValues)`
2. Added columnTransformationUdfName parameter to allow caller to indicate a UDF to be used at the column level

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/DigestGenerationHandler.java [hunk 1]
```diff
@@ -106,6 +106,7 @@ else if (value instanceof StagedFilesFieldValue)
             .addAllValues(filteredStagingFieldValues)
             .addAllFieldTypes(filteredStagingFieldTypes)
             .putAllTypeConversionUdfNames(udfBasedDigestGenStrategy.typeConversionUdfNames())
+            .columnUdf(udfBasedDigestGenStrategy.columnTransformUdfName())
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/DigestGenerationHandler.java:109]: let's call it: columnTransformationUdfName every where

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/values/DigestUdfAbstract.java [hunk 2]
```diff
@@ -40,4 +41,6 @@ public interface DigestUdfAbstract extends Value
     List<FieldType> fieldTypes();
 
     Map<DataType, String> typeConversionUdfNames();
+
+    Optional<String> columnUdf();
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/values/DigestUdfAbstract.java:45]: columnTransformationUdfName()

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/UDFBasedDigestGenStrategyAbstract.java [hunk 3]
```diff
@@ -38,6 +39,8 @@ public interface UDFBasedDigestGenStrategyAbstract extends DigestGenStrategy
 
     Map<DataType, String> typeConversionUdfNames();
 
+    Optional<String> columnTransformUdfName();
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/UDFBasedDigestGenStrategyAbstract.java:42]: columnTransformationUdfName

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/DigestGenerationHandler.java (hunk 1 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/DigestGenerationHandler.java]: Sure changed!

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/values/DigestUdfAbstract.java (hunk 2 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/values/DigestUdfAbstract.java]: Sure changed!

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/UDFBasedDigestGenStrategyAbstract.java (hunk 3 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/digest/UDFBasedDigestGenStrategyAbstract.java]: Sure changed!

**Review by prasar-ashutosh — APPROVED** [review]

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**